src/evo/creditpool.cpp

```diff
@@ -249,14 +249,12 @@ bool CCreditPoolDiff::SetTarget(const CTransaction& tx, const CAmount blockRewar
 
 bool CCreditPoolDiff::Lock(const CTransaction& tx, TxValidationState& state)
 {
-    CAssetLockPayload assetLockTx;
-    if (!GetTxPayload(tx, assetLockTx)) {
+    if (CAssetLockPayload assetLockTx; !GetTxPayload(tx, assetLockTx)) {
         return state.Invalid(TxValidationResult::TX_CONSENSUS, "failed-creditpool-lock-payload");
     }
 
     for (const CTxOut& txout : tx.vout) {
-        const CScript& script = txout.scriptPubKey;
-        if (script.empty() || script[0] != OP_RETURN) continue;
+        if (const CScript& script = txout.scriptPubKey; script.empty() || script[0] != OP_RETURN) continue;
```

Comment:
script is not used further; just to make code more tidy. 

But inside `if` it looks ugly IMO, let's simplify then:

```suggestion
        if (txout.scriptPubKey.empty() || txout.scriptPubKey[0] != OP_RETURN) continue;
```